app/code/community/Bolt/Boltpay/Model/FeatureSwitch.php

```diff
@@ -48,4 +58,15 @@ public function updateSwitchesFromBolt()
         Mage::getModel('core/config')->saveConfig('payment/boltpay/featureSwitches', json_encode($switches));
         Mage::getModel('core/config')->cleanCache();
     }
+
+    public function updateSwitchesFromBoltIfNeeded()
```

Comment:
I think we can just move this needUpdateFeatures check to updateSwitchesFromBolt and get rid of this method.

Reply:
Down the road, we will use method **updateSwitchesFromBolt**, when we handle API call from bolt server about we need update feature switches.
https://app.asana.com/0/1163213880723410/1163214273078115/f

Reply:
@mccottry We need to update feature switches from bolt in two cases:
- when we install/update bolt
- when bolt send to us special API call that server has new feature switches.

We will use updateSwitchesFromBolt in the second case.

Reply:
@vitaliyreznikov See Leon's comment below. We can keep the updateFeatureSwitches method the same and move the "is needed" part to the observer.

Reply:
Got it, thank you for an explanation. Done.